apps/meteor/ee/app/models/server/models/LivechatUnit.js

```diff
@@ -105,6 +106,8 @@ export class LivechatUnit extends LivechatDepartment {
 			);
 		});
 
+		Promise.await(LivechatRooms.associateRoomsWithDepartmentToUnit(departmentsToSave, _id));
+
 		return _.extend(record, { _id });
```

Comment:
```suggestion
		return { _id, ...record }
```

Reply:
Sorry but I had to revert this back :disappointed:  because this is causing some weird issues in some endpoints defined here because some of the helper files are still in js and the typescript isn't able to correctly resolve its types causing issues -> `apps/meteor/ee/app/livechat-enterprise/server/api/units.ts`. Will check this again when we attempt to covert LivechatUnit model to typescript

Reply:
Oh, if it was a "typings" problems, you can use Object.assign, and that should work 👀

Reply:
This should be fixed now :smile: